library/swift/src/filters/FilterDataStatus.swift

```diff
@@ -21,4 +21,17 @@ public enum FilterDataStatus {
   /// This should be called by filters which must parse a larger block of the incoming data before
   /// continuing processing.
   case stopIterationAndBuffer(Data)
+
+  /// Do not iterate to any of the remaining filters in the chain, and do not internally buffer
+  /// data.
+  ///
+  /// This filter will continue to be called with new chunks of data.
+  ///
+  /// Returning `continue` from `onRequestData()`/`onResponseData()` or calling
+  /// `continueRequest()`/`continueResponse()` MUST be called when continued filter iteration is
+  /// desired.
+  ///
+  /// This may be called by filters which must parse a larger block of the incoming data before
+  /// continuing processing, and will handle their own buffering.
+  case stopIterationNoBuffer
```

Comment:
We had actually dropped this deliberately since it wasn't something we thought was needed yet. What was the reasoning for adding it back? If we want to keep it, we should also add it to the Android interfaces

Reply:
We discussed earlier, but posting here for posterity - leaving this in potentially eases the path for migrating over platform logic that already does its own buffering, since less code needs to change initially. It also still provides for application logic that for one reason or another must to do its own buffering.